.changeset/metal-bugs-beg.md

```diff
@@ -0,0 +1,9 @@
+---
+"@comet/admin": minor
+---
+
+Add the ability to show and hide individual columns depending on the screen size when using `DataGridPro` with `usePersistentColumnState`
+
+A compact view of a grid can be created by defining additional columns that show a combination of others.
+Then, you can use the `showOnlyInView` setting on certain columns to show them only in the default or compact view.
+When defining the columns, use the `GridColDef` type from `@comet/admin` instead of `@mui/x-data-grid`.
```

Comment:
Should we add an ESLint import restriction for this?

Reply:
Maybe... not sure if there would still be good use-cases for the MUI type 🤔 

I'd like to wait for the discussion on using MUIs actions column type from https://github.com/vivid-planet/comet/pull/2044. 
If we use the actions column, we need a different (likely custom) type for the columns in that case as well.

Reply:
Since we don't use the actions column, what's your opinion on this now?

Reply:
> Should we add an ESLint import restriction for this?

Yes, I will add this 👍🏼